### ISSUE TITLE
Adding ServiceBehaviorAttribute to ServiceHostBase

### ISSUE THREAD

**Issue by oruchreis** [CoreWCF/CoreWCF#686]
Hi,
I'm trying to add a default ServiceBehaviorAttribute to all services. I achieved this by removing the default and adding a new one via ServiceHost.Description.Behaviors in old WCF. When I do this in coreWCF like that I got `InvalidOperationException`:
```csharp
service
  .ConfigureServiceHostBase(serviceMetaData.Type, hostBase =>
  {
      hostBase.Description.Behaviors.Remove<ServiceBehaviorAttribute>();
      hostBase.Description.Behaviors.Add(new ServiceBehaviorAttribute
      {
          Namespace = Constants.NamespaceBase + "service/",
          ConcurrencyMode = ConcurrencyMode.Single,
          InstanceContextMode = InstanceContextMode.PerCall,
      });
```
This leads to this exception:
```
InvalidOperationException: The service type provided could not be loaded as a service because it does not have a default (parameter-less) constructor. To fix the problem, add a default constructor to the type, or pass an instance of the type to the host.

    CoreWCF.Dispatcher.InstanceBehavior.GetInstance(InstanceContext instanceContext, Message request)
    CoreWCF.InstanceContext.GetServiceInstance(Message message)
    CoreWCF.Dispatcher.InstanceBehavior.EnsureServiceInstance(MessageRpc rpc)
    CoreWCF.Dispatcher.ImmutableDispatchRuntime.ProcessMessageAsync(MessageRpc rpc)
    CoreWCF.Dispatcher.MessageRpc.ProcessAsync(Boolean isOperationContextSet)
```
But all of my service types have parameterless default constructor.
If I don't remove the default one and change its contents then it succeeds without any error:
```csharp
service
    .ConfigureServiceHostBase(serviceMetaData.Type, hostBase =>
    {
        var serviceBehaviorAttr = ((ServiceBehaviorAttribute)hostBase.Description.Behaviors[typeof(ServiceBehaviorAttribute)]);
        serviceBehaviorAttr.Namespace = Constants.NamespaceBase + "service/";
        serviceBehaviorAttr.ConcurrencyMode = ConcurrencyMode.Single;
        serviceBehaviorAttr.InstanceContextMode = InstanceContextMode.PerCall;
```

**Comment by mconnew** [comment]
Does your service type have a default constructor? The automatically provided ServiceBehaviorAttribute has an `IInstanceProvider` configured which can pull the instance from DI when it doesn't have a default constructor. By replacing the behavior, you end up with the it doing the default when a provider hasn't been set which is to look for a default constructor.

**Comment by oruchreis** [comment]
Yes, actually all of my service types doesn't implement any parametered or parameterless consturctor at all, which means they have a parameterless default constructor. And when I debugged into the source, the instance provider is null as you mentioned, and `hasDefaultConstructor` is true in here:
https://github.com/CoreWCF/CoreWCF/blob/7ef9498f66811fa7b678e5a4c0091d72d101012a/src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/InstanceBehavior.cs#L31

**Comment by mconnew** [comment]
I think I found the bug. The if statement around the code at https://github.com/CoreWCF/CoreWCF/blob/main/src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/InstanceBehavior.cs#L45-L52 is inverted. It's saying if there isn't a default constructor, then create a delegate which constructs an instance from the default constructor. The `!` needs to be removed so it runs if there is a default constructor.

**Comment by mconnew** [comment]
@oruchreis, would you be able to submit a test which validates this scenario and make the code change? It's okay if you can't, it will get fixed at some point.
